src/diffusers/models/transformer_2d.py

```diff
@@ -213,22 +213,28 @@ def __init__(
 
     def forward(
         self,
-        hidden_states,
-        encoder_hidden_states=None,
-        timestep=None,
-        class_labels=None,
-        cross_attention_kwargs=None,
+        hidden_states: Tensor,
+        attention_mask: Optional[Tensor] = None,
+        encoder_hidden_states: Optional[Tensor] = None,
+        encoder_attention_mask: Optional[Tensor] = None,
+        timestep: Optional[LongTensor] = None,
```

Comment:
can we leave this at second position? Otherwise this is a breaking change, e.g. many people do:

prior(x, t)

assuming that timesteps is at second position :-)

Reply:
moved new params to ends of method signatures, to preserve positional param interface.  
though I left `return_dict` as final argument, because `create_custom_forward` regards it as a special-case.
I did notice some questionable binding of `checkpoint()` arguments in unet_2d_blocks (`cross_attention_kwargs` arg bound to `timestep` parameter's position), so I assigned `None`s to `timestep` and to `class_labels` to fill the gap.

Reply:
I agree with the latest change and keeping `return_dict` last. It's awesome that you unsurfaced that error in `checkpoint()`, I had not read this comment yet and it took me a while to see what was going on, so I suggested to add comments to the `None` positional arguments for clarity.